### ISSUE TITLE
Standardize script execution when appending elements to the DOM

### ISSUE THREAD

**Issue by victorg1991** [liferay/liferay-frontend-guidelines#38]
In some places in the portal, when doing an Ajax request the response contains scripts apart from other HTML elements, simply setting the innerHTML property of an element doesn't execute those scripts. In order to solve this, the parse-content AUI module was used

Now that we don't want that option, there are a couple of options:

- Using `globalEval.runScriptsInElement` from 'metal-dom' [ref](https://github.com/metal/metal.js/blob/17e5fb69956a5c14d367addc1af0898a89cff42e/packages/metal-dom/src/globalEval.js#L92)
- Using `range.createContextualFragment` as [here](https://github.com/brianchandotcom/liferay-portal/blob/8d3eedcbbadd2b1a02e81c0f9820bc739f84e19e/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/side_navigation.es.js#L848-L857)

My main concern about the second one is that the [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Range/createContextualFragment) marks it as experimental

**Comment by wincent** [comment]
I'm not super worried about the "experimental" status. If you look at the MDN table, you'll see it is *very* broadly implemented, so I'm pretty skeptical that it would change. Also, looking at the [revision history in the spec](https://w3c.github.io/DOM-Parsing/#revision-history), there seem to have been only very minor changes in the past. As far as I can tell, it's a [pretty widely recommended](https://stackoverflow.com/questions/9284117/inserting-arbitrary-html-into-a-documentfragment) approach.

The other question though is do we care about possible lack of support in Edge Mobile and Samsung Internet? It's not clear to me whether support is *known* to be missing there, or simply that the level of support isn't known.

**Comment by victorg1991** [comment]
I think we don't support those browsers, at least they are not listed in the compatibility matrix

Is it correct @jbalsas?

**Comment by jbalsas** [comment]
Yeah, technically we don't need to support those browsers. We should build some thin wrapper around this, but sounds good to me :)
